src/middleware.rs

```diff
@@ -384,9 +428,128 @@ pub enum OperationArg {
     Key(AnchoredKey),
 }
 
+impl OperationArg {
+    pub fn is_none(&self) -> bool {
+        matches!(self, Self::None)
+    }
+    pub fn statement(&self) -> Result<Statement> {
+        match self {
+            Self::Statement(statement) => Ok(statement.clone()),
+            _ => Err(anyhow!("Operation argument {:?} is not a statement.", self)),
+        }
+    }
+    pub fn key(&self) -> Result<AnchoredKey> {
+        match self {
+            Self::Key(ak) => Ok(ak.clone()),
+            _ => Err(anyhow!("Operation argument {:?} is not a key.", self)),
+        }
+    }
+}
+
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct Operation(pub NativeOperation, pub Vec<OperationArg>);
 
+impl Operation {
+    pub fn code(&self) -> NativeOperation {
+        self.0
+    }
+    pub fn args(&self) -> &[OperationArg] {
+        &self.1
+    }
+    // TODO: Argument checking.
+    /// Checks the given operation against a statement.
+    pub fn check(&self, output_statement: Statement) -> Result<bool> {
+        use NativeOperation::*;
+        match self.0 {
+            // Nothing to check.
+            None => Ok(output_statement.code() == NativeStatement::None),
+            // Check that the resulting statement is of type `ValueOf`
+            // and its origin is `SELF`.
+            NewEntry =>
+                Ok(output_statement.code() == NativeStatement::ValueOf && output_statement.args()[0].key()?.origin() == SELF)
+            ,
+            // Check that the operation acts on a statement *and* the
+            // output is equal to this statement.
+            CopyStatement => Ok(output_statement == self.args()[0].statement()?)
```

Comment:
This is a design question: In the Devcon implementation Operation and Statement were more strongly typed:
https://github.com/0xPARC/parcnet/blob/f7ef5e2fedea10f349c6343fb24d84374edbbad2/pod2/src/pod/operation.rs#L17-L33
https://github.com/0xPARC/parcnet/blob/f7ef5e2fedea10f349c6343fb24d84374edbbad2/pod2/src/pod/statement.rs#L79-L90

Here they are very generic, where the arguments are not strongly typed to be correct.
This generic code leads to a more verbose `check` implementation, but I also think it has benefits in other parts of the code.

Do you have any thoughts on this approach VS the previous one?  Do you like this implementation of `check`?